modules/kubernetes/ingress-nginx/variables.tf

```diff
@@ -27,38 +15,53 @@ variable "default_certificate" {
   }
 }
 
-variable "linkerd_enabled" {
-  description = "Should linkerd be enabled"
-  type        = bool
-  default     = false
+variable "external_dns_hostname" {
+  description = "Hostname for external-dns to use"
+  type        = string
+  default     = ""
 }
 
-variable "datadog_enabled" {
-  description = "Should datadog be enabled"
+variable "public_private_enabled" {
+  description = "Should ingress controllers for both public and private be created?"
   type        = bool
   default     = false
 }
 
-variable "extra_config" {
-  description = "Extra config to add to Ingress NGINX"
-  type        = map(string)
-  default     = {}
+variable "config" {
+  type = object({
+    allow_snippet_annotations = optional(bool, false)
+    http_snippet              = optional(string, "")
+    extra_config              = optional(map(string), {})
+    extra_headers             = optional(map(string), {})
+  })
+}
+
+variable "config_public" {
```

Comment:
`config_public` is not a great name. Does anyone have a better idea for the name?

Reply:
Not sure, but maybe `private_ingress_customization` and `public_ingress_customization`?

Reply:
... or `extra_{private,public}_customization` given that "extra" is a thing in this context?

Reply:
Customization may be a good word. The issue comes from the fact that we went from single attributes to duplicating them in multiple objects to enable global and override in a sane manner. So `config` will be used if `private-public` is false and also as a global default if `private-public` is true. Then the other two blocks enable specific overrides for the public and private ingress controllers.